eos/scattering/ee-to-ccbar.cc

```diff
@@ -0,0 +1,673 @@
+                        break;
+                    case 13:
+                    case 14:
+                        channel_array[i] = std::make_shared<VV_Pwavechan<nchannels, nresonances, order>>(channel_names[i] + "_chan", m_Dst0, m_Dst0, 1, _get_g0_column(_filter_channel_index(i), std::make_index_sequence<nresonances>()));
+                        break;
+                    case 15:
+                        channel_array[i] = std::make_shared<VV_Fwavechan<nchannels, nresonances, order>>(channel_names[i] + "_chan", m_Dst0, m_Dst0, 3, _get_g0_column(_filter_channel_index(i), std::make_index_sequence<nresonances>()));
+                        break;
+                    case 16:
+                    case 17:
+                        channel_array[i] = std::make_shared<VV_Pwavechan<nchannels, nresonances, order>>(channel_names[i] + "_chan", m_Dst, m_Dst, 1, _get_g0_column(_filter_channel_index(i), std::make_index_sequence<nresonances>()));
+                        break;
+                    case 18:
+                        channel_array[i] = std::make_shared<VV_Fwavechan<nchannels, nresonances, order>>(channel_names[i] + "_chan", m_Dst, m_Dst, 3, _get_g0_column(_filter_channel_index(i), std::make_index_sequence<nresonances>()));
+                        break;
+                    case 19:
+                    case 20:
+                        channel_array[i] = std::make_shared<VP_Pwavechan<nchannels, nresonances, order>>(channel_names[i] + "_chan", m_Ds, m_Dsst, 1, _get_g0_column(_filter_channel_index(i), std::make_index_sequence<nresonances>()));
+                        break;
+                    case 21:
+                    case 22:
+                        channel_array[i] = std::make_shared<VV_Pwavechan<nchannels, nresonances, order>>(channel_names[i] + "_chan", m_Dsst, m_Dsst, 1, _get_g0_column(_filter_channel_index(i), std::make_index_sequence<nresonances>()));
+                        break;
+                    case 23:
+                        channel_array[i] = std::make_shared<VV_Fwavechan<nchannels, nresonances, order>>(channel_names[i] + "_chan", m_Dsst, m_Dsst, 3, _get_g0_column(_filter_channel_index(i), std::make_index_sequence<nresonances>()));
+                        break;
+
+                    default:
+                        throw InternalError("The number of requested channels (array) is larger than the number of channel names.");
+                }
+            }
+
+            K = std::shared_ptr<KMatrix<nchannels, nresonances, order>> (
+                new KMatrix<nchannels, nresonances, order>(
+                    channel_array,
+                    resonance_array,
+                    _get_bkgpol(std::make_index_sequence<order + 1>(), std::make_index_sequence<nchannels>(), std::make_index_sequence<nchannels>()),
+                    "KMatrix"
```

Comment:
Maybe this requires a more helpful name than "KMatrix"?

Reply:
`"e^+e^-->ccbar::KMatrix"` ?

Reply:
Why not just ``e^+e^-->ccbar`` Nobody will dispute its character as a Kmatrix, will they?